embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/LogSinkImpl.kt

```diff
@@ -21,15 +21,16 @@ internal class LogSinkImpl : LogSink {
     }
 
     override fun completedLogs(): List<EmbraceLogRecordData> {
-        synchronized(storedLogs) {
-            return storedLogs.toList()
-        }
+        return storedLogs.toList()
     }
 
-    override fun flushLogs(): List<EmbraceLogRecordData> {
-        synchronized(storedLogs) {
-            val flushedLogs = storedLogs.toList()
-            storedLogs.clear()
+    override fun flushLogs(max: Int?): List<EmbraceLogRecordData> {
+        synchronized(flushLock) {
+            val maxIndex = max?.let {
+                minOf(storedLogs.size, it)
+            } ?: storedLogs.size
+            val flushedLogs = storedLogs.toList().subList(0, maxIndex)
```

Comment:
Since the list is ordered, is it possible just to do `storedLogs.take(maxIndex)` for simplicity and efficiency?

Reply:
Nice... updated !!